lib/internal/webstreams/adapters.js

```diff
@@ -457,7 +459,10 @@ function newReadableStreamFromStreamReadable(streamReadable, options = kEmptyObj
   streamReadable.on('data', onData);
 
   return new ReadableStream({
-    start(c) { controller = c; },
+    start(c) {
+      controller = c;
+      controller[kFresh] = true;
+    },
 
     pull() { streamReadable.resume(); },
```

Comment:
Then we should probably not call `streamReadable.resume` in the `pull` method

Reply:
Yes this was tried also i think you can check some of the earlier PRs best strategy seems like would be changing the adapters logic somehow instead of resume if we could consume the data one by one

Reply:
I know what's going on here, the Readable is always in byte mode, 
```js
  readableHighWaterMark: {
    __proto__: null,
    enumerable: false,
    get: function() {
      return this._readableState.highWaterMark;
    },
  },
```
and this returns size in bytes, but whatwg stream defaults to non-byte mode, which want HWM size in chunks

Reply:
https://streams.spec.whatwg.org/#high-water-mark:~:text=Concretely%2C%20a%20queuing,floating%2Dpoint%20units.